### ISSUE TITLE
Feature Request: Separate TF Resource for managing databases of a DBaaS instance

### ISSUE THREAD

**Issue by tstrohmeier** [exoscale/terraform-provider-exoscale#136]
Hello,

It would be nice to have different resources for database instances and the databases itself. 
This would allow us to manage also the databases with Terraform directly.

### Example:

```hcl
locals {
  zone = "ch-dk-2"
}

resource "exoscale_database_instance" "pg_prod" {
  zone = local.zone
  name = "pg-prod"
  type = "pg"
  plan = "startup-4"

  maintenance_dow  = "sunday"
  maintenance_time = "23:00:00"

  termination_protection = true

  pg {
    version = "13"
    backup_schedule = "04:00"
  }
}

resource "exoscale_database" "my_database" {
    database_instance = exoscale_database_instance.pg_prod
    database_name = "<DATABASE_NAME>"
}

```

### Links from other projects:
* https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/pg_database <del>https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/database </del>
* https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database

**Comment by retrack** [comment]
Thanks for the suggestion, we will take it in consideration in our planning and also welcome PRs as well.

**Comment by tstrohmeier** [comment]
@retrack What is the status on this feature request?

**Comment by smangelkramer** [comment]
@retrack Any updates on this?

**Comment by SebiSpace** [comment]
@retrack Hi, this issue is now open since more than a year and you said you would take it into consideration, did you already have a look at it?

**Comment by llambiel** [comment]
> @retrack Hi, this issue is now open since more than a year and you said you would take it into consideration, did you already have a look at it?

Hi @SebiSpace thanks for the heads-up.

It's still something we are considering, however it's not on top of the priority list at the moment. Unfortunately we don't have a more concrete timeline to share right now but will do once we have.

**Comment by tstrohmeier** [comment]
It seams like aiven (the provider used by Exoscale) is already offering this: 
https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/pg and https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/pg_database

@llambiel  So I guess the API from aiven is available and could be used in a **Exoscale** TF resource?

**Comment by llambiel** [comment]
> It seams like aiven (the provider used by Exoscale) is already offering this: https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/pg and https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/pg_database
> 
> @llambiel So I guess the API from aiven is available and could be used in a TF resource?

You cannot use the Aiven API in direct to manage Exoscale DBaaS resources. If you do so you'll have to setup an account with them. You won't be able to deploy the resources on top of Exoscale that way.

**Comment by tstrohmeier** [comment]
@llambiel Yes I know. I mean with "available" that aiven has implemented the API on there side and that it should be possible for Exoscale to implement this API in a reasonable time?

I mean this ticket is open for over one year now.

**Comment by llambiel** [comment]
@tstrohmeier Actually support of this feature requires more than just the Terraform implementation. Some prior work on our API has to be made before we could start with the Terraform implementation. Such work has been added in the pipe. We'll revert back to you once we have an update on this front.

**Comment by tstrohmeier** [comment]
@llambiel   Ok I see. Can you please keep us here in the ticket updated about the progress?

**Comment by simisoft-exo** [comment]
@tstrohmeier  We have started work to enable this feature in the API

**Comment by tstrohmeier** [comment]
@simisoft-exo This sounds good! What is the process on it?

**Comment by simisoft-exo** [comment]
> @simisoft-exo This sounds good! What is the process on it?

The feature is available in our API.
cc: @ste-m @kobajagi @jessicatoscani @llambiel

**Comment by simisoft-exo** [comment]
@tstrohmeier  The feature is also available in the TF provider version v0.64.1
